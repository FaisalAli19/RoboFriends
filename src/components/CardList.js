import React from 'react'

import Card from './Card'

const CardList = ({ robots }) => {
	const cardArray = robots.map((robot, i) => {
		return (
			<Card 
				key={i} 
				id={robot.id} 
				name={robot.name} 
				email={robot.email}
			/>
	);
	})
	return (
		<div>
			{ cardArray }
		</div>
	);
}

export default CardList;