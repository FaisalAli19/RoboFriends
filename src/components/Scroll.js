import React from 'react'
import { connect } from 'react-redux';

const Scroll = (props) => {
	let children;
	if (props.isPending) children = <h1 style={{'fontFamily': 'SEGA LOGO FONT', 'fontWeight': '200', color:' #0ccac5'}}>Loading...</h1>
	else children = props.children
	return (
		<div style={{overflowY: 'auto', border: '1px solid #000', height: '600px'}}>
			{children}
		</div>
	);
}

const mapStateToProps = ({ requestsRobots }) => {
	return {
		isPending: requestsRobots.isPending
	};
}

export default connect(mapStateToProps)(Scroll);