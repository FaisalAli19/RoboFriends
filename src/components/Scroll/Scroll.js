import React from 'react';

const Scroll = (props) => {
	let children;
	if (props.isPending) children = <h1 className="loading" style={{'fontFamily': 'SEGA LOGO FONT', 'fontWeight': '200', color:' #0ccac5'}}>Loading...</h1>
	else children = props.children
	return (
		<div style={{overflowY: 'auto', border: '1px solid #000', height: '550px'}}>
			{children}
		</div>
	);
}

export default Scroll;