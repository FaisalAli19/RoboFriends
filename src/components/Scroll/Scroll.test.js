import React from 'react';
import { shallow, mount } from 'enzyme'

import Scroll from './Scroll';
import CardList from '../CardList/CardList'

describe('Scroll Component', () => {
	it('should render correctly', () => {
		expect(shallow(<Scroll />)).toMatchSnapshot()
	});
	it('should render is loading', () => {
		let wrapper = mount(<Scroll isPending={true} />)
		expect(wrapper.find('.loading').length).toEqual(1)
	});
});