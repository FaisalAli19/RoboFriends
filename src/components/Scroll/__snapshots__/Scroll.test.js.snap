// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll Component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scroll />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "style": Object {
        "border": "1px solid #000",
        "height": "550px",
        "overflowY": "auto",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "style": Object {
          "border": "1px solid #000",
          "height": "550px",
          "overflowY": "auto",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
